src/advanced_agent/learning/prompt_manager.py

```
@@ -0,0 +1,909 @@
+from typing import Dict, Any, List, Optional, Union, Tuple
+from datetime import datetime
+from pathlib import Path
+import hashlib
+import uuid
+
+from ..reasoning.ollama_client import OllamaClient
+from ..database.models import PromptTemplate
+from ..database.connection import get_database_manager
+
+
+class PromptTemplate:
```

Ruff (0.12.2)

Redefinition of unused `PromptTemplate` from line 18

(F811)